app/controllers/prx_auth/rails/sessions_controller.rb

```diff
@@ -70,6 +70,7 @@ def refresh
       sign_out_user
       session[WILDCARD_SESSION_KEY] = wildcard
 
+      set_after_sign_in_path(request.referer.presence || "/")
```

Comment:
And here's the 2nd place to set it... admins request wildcard access by clicking a link `/auth/sessions/refresh?wildcard=true`.  So in this case, we grab the referrer, and send them back there.